### PR TITLE
add R.forEachObjIndexed to iterate over an object

### PR DESCRIPTION
**Pull request by centsent**
Discussion here: #1893

### CODE REVIEW

File: src/forEachObjIndexed.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @return {Object} The original object.
+ * @example
+ *
+ *      var printKeyConcatValue = (value, key) => console.log(key + ':' + value);
+ *      R.forEachObjIndexed(printKeyConcatValue, {x: 1, y: 2}); //=> {x: 1, y: 2}
+ *      // logs x:1
+ *      // logs y:2
+ * @symb R.forEachObjIndexed(f, {x: a, y: b}) = {x: a, y: b}
+ */
+module.exports = _curry2(_checkForMethod('forEachObjIndexed', function forEachObjIndexed(fn, obj) {
```
Comment (davidchambers) [inline src/forEachObjIndexed.js]: I don't think we should dispatch to a `forEachObjIndexed` method.

Comment (CrossEye) [inline src/forEachObjIndexed.js]: Agreed.

**Comment by CrossEye** [comment]
I don't feel much of a need for this function, but I can see why someone might want it.

A tepid :herb: from me.

**Comment by davidchambers** [comment]
I'd prefer to deprecate `R.forEach`, but so long as that function exists it seems completely reasonable to me to include the "indexed" equivalent.

File: src/forEachObjIndexed.js (hunk 1 above)
Comment (centsent) [inline src/forEachObjIndexed.js]: Updated, review it again please.

File: src/forEachObjIndexed.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Iterate over an input `object`, calling a provided function `fn` for each
+ * key and value in the object.
+ *
+ * `fn` receives three argument: *(value, key, obj)*.
+ *
+ * @func
+ * @memberOf R
+ * @category Object
+ * @sig f => (a -> b) -> f a -> f b
```
Comment (davidchambers) [inline src/forEachObjIndexed.js]: I think the signature should be something like this:

```haskell
forEachObjIndexed :: ((a, String, StrMap a) -> Any) -> StrMap a -> StrMap a
```

File: test/forEachObjIndexed.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    eq(sideEffect, obj);
+  });
+
+  it('returns the original object', function() {
+    var s = '';
+    eq(R.forEachObjIndexed(function(value) { s += value; }, obj), obj);
+    eq('12123', s);
+  });
+
+  it('no dispatches to `forEachObjIndexed` method', function() {
```
Comment (davidchambers) [inline test/forEachObjIndexed.js]: I don't think it's necessary to assert this.

Comment (centsent) [inline test/forEachObjIndexed.js]: Removed, check it again :)

**Comment by davidchambers** [comment]
Looks okay to me. :)

**Pull request merged**